packages/manager/modules/pci/src/projects/project/storages/components/password-clipboard/template.html

```diff
@@ -1,26 +1,33 @@
-<oui-field
-    class="d-inline"
-    data-label="{{:: $ctrl.fieldSecretKeyLabel }}"
->
-    <div class="mb-0 oui-input-overlay oui-clipboard" data-ng-init="passwordClipboard = {isText: false}">
+<oui-field class="d-inline" data-label="{{:: $ctrl.fieldSecretKeyLabel }}">
+    <div class="mb-0 oui-input-overlay oui-clipboard">
         <input
             class="oui-input oui-clipboard__control"
             data-ng-attr-id="{{:: $ctrl.SECRET_KEY_INPUT_ID}}"
             data-ng-attr-name="{{:: $ctrl.SECRET_KEY_INPUT_ID}}"
-            data-ng-attr-type="{{ passwordClipboard.isText ? 'text' : 'password' }}"
+            data-ng-attr-type="{{ $ctrl.shouldDisplayPasswordInput? 'text' : 'password' }}"
             data-ng-value="$ctrl.fieldSecretKeyValue"
             data-ng-click="$ctrl.onClipboardClick()"
             data-oui-tooltip="{{ $ctrl.model.tooltipText }}"
             readonly
         />
         <button
             type="button"
+            data-ng-if="!$ctrl.shouldDisplayPasswordInput"
             class="oui-button mr-4"
-            aria-label="{{ $ctrl.getPasswordLabel(passwordClipboard.isText) }}"
-            data-ng-click="passwordClipboard.isText = !passwordClipboard.isText;"
+            aria-label="{{ $ctrl.getPasswordLabel($ctrl.shouldDisplayPasswordInput) }}"
+            data-ng-click="$ctrl.shouldDisplayPasswordInput = !$ctrl.shouldDisplayPasswordInput"
+        >
+            <span class="oui-icon oui-icon-eye-open" aria-hidden="true"></span>
+        </button>
+        <button
+            type="button"
+            data-ng-if="$ctrl.shouldDisplayPasswordInput"
+            class="oui-button mr-4"
+            aria-label="{{ $ctrl.getPasswordLabel($ctrl.shouldDisplayPasswordInput) }}"
+            data-ng-click="$ctrl.shouldDisplayPasswordInput = !$ctrl.shouldDisplayPasswordInput"
         >
             <span
-                class="oui-icon oui-icon-eye-open"
+                class="oui-icon oui-icon-eye-closed"
                 aria-hidden="true"
             ></span>
         </button>
```

Comment:
**nitpick:** Since only the icon is different, we can use `ng-class` instead of maintaining 2 buttons

Reply:
Done Anoop